### PR TITLE
Don't instantiate template for low DIM counts, to fix a warning

### PR DESCRIPTION
**Pull request by manopapad**
The warning was

```
    In file included from /Users/mpapadakis/cunumeric/src/cunumeric/matrix/batched_cholesky.cc:19:
    /Users/mpapadakis/cunumeric/src/cunumeric/matrix/batched_cholesky_template.inl:82:9: warning: array index -1 is before the beginning of the array [-Warray-bounds]
        if (in_strides[DIM - 2] != ncols || in_strides[DIM - 1] != 1) {
            ^          ~~~~~~~
    /Users/mpapadakis/legate.core/src/core/utilities/dispatch.h:34:27: note: in instantiation of function template specialization 'cunumeric::BatchedCholeskyImpl<cunumeric::VariantKind::CPU>::operator()<legate::Type::Code::BOOL, 1>' requested here
            return f.template operator()<Type::Code::BOOL, DIM>(std::forward<Fnargs>(args)...);
                              ^
    /Users/mpapadakis/cunumeric/src/cunumeric/matrix/batched_cholesky.cc:74:3: note: in instantiation of function template specialization 'cunumeric::batched_cholesky_task_context_dispatch<cunumeric::VariantKind::CPU>' requested here
      batched_cholesky_task_context_dispatch<VariantKind::CPU>(context);
      ^
    /Users/mpapadakis/cunumeric/src/cunumeric/matrix/batched_cholesky_template.inl:78:5: note: array 'in_strides' declared here
        size_t in_strides[DIM];
        ^
    /Users/mpapadakis/cunumeric/src/cunumeric/matrix/batched_cholesky_template.inl:89:9: warning: array index -1 is before the beginning of the array [-Warray-bounds]
        if (out_strides[DIM - 2] != ncols || out_strides[DIM - 1] != 1) {
            ^           ~~~~~~~
    /Users/mpapadakis/cunumeric/src/cunumeric/matrix/batched_cholesky_template.inl:79:5: note: array 'out_strides' declared here
        size_t out_strides[DIM];
        ^
```

### CODE REVIEW

File: src/cunumeric/matrix/batched_cholesky_template.inl [hunk 1]
```diff
@@ -57,7 +57,7 @@ struct _cholesky_supported {
 
 template <VariantKind KIND>
 struct BatchedCholeskyImpl {
-  template <Type::Code CODE, int DIM>
+  template <Type::Code CODE, int32_t DIM, std::enable_if_t<2 < DIM>* = nullptr>
```
Comment (jjwilke) [inline src/cunumeric/matrix/batched_cholesky_template.inl:60]: nit: `DIM > 2` is more readable and matches the other enable_if

**Review by jjwilke — APPROVED** [review]
LGTM

File: src/cunumeric/matrix/batched_cholesky_template.inl (hunk 1 above)
Comment (manopapad) [reply src/cunumeric/matrix/batched_cholesky_template.inl]: Yup, I convinced the compiler to accept it by adding parens dd5e01d

**Comment by manopapad** [comment]
The CI failures are unrelated, will open a fix soon.

**Pull request merged**